### PR TITLE
Fixup role match in classifier of resource_group

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

https://github.com/StarRocks/starrocks/issues/4381
## Problem Summary(Required) ：

When matching roleName to role condition of resource group classifier, use simple role name instead of qualified role name.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|0|3|00.00%|

**Review by laotan332 — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupMgr.java [hunk 1]
```diff
@@ -119,8 +119,11 @@ public void createWorkGroup(CreateWorkGroupStmt stmt) throws DdlException {
                 //default_cluster:test
                 String[] userParts = qualifiedUser.split(":");
                 String user = userParts[userParts.length - 1];
-                String roleName = Catalog.getCurrentCatalog().getAuth()
+                String qualifiedRoleName = Catalog.getCurrentCatalog().getAuth()
                         .getRoleName(ConnectContext.get().getCurrentUserIdentity());
+                //default_cluster:role
+                String[] roleParts = qualifiedRoleName.split(":");
+                String roleName = roleParts[roleParts.length - 1];
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupMgr.java:126]: here could throw  ArrayIndexOutOfBoundsException?

Comment (satanson) [reply fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupMgr.java:126]: NPE may be thrown, 
ArrayIndexOutOfBoundsException is impossible;
for an example:
1. when qualifiedRoleName="default_cluster:role1", split  into two parts, so roleParts.length-1 = 1; safe
2.  when qualifiedRoleName="role1", split into one part, roleParts.length-1 = 0; safe
3. qualifiedRoleName maybe null, so throw NPE.

**Review by kangkaisen — DISMISSED** [review]

**Review by ZiheLiu — DISMISSED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 7 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|0|7|00.00%|

**Review by laotan332 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**